### PR TITLE
avidemux: rewrite derivation

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

This drastically reduces the complexity of the `avidemux` derivation
and adds QT5 support (see #33248).

Rather than invoking `cmake` over preconfigured hooks, it's much easier
to use the `bootStrap.bash` script provided by the developers to do the
installation tasks. Furthermore this script makes it way easier to
configure which parts of `avidemux` should be used (e.g. CLI-only) or
without the plugins.

In order to create a CLI-only instance you can simply override the
derivation:

```
avidemux.override {
  withQT = false;
}
```

It's possible to set the default executable as well (`avidemux` creates
a `avidemux_qt5` and `avidemux_cli` executable by default):

```
avidemux.override {
  default = "cli"; # default is `qt5`
}
```

The GTK support has been dropped entirely since it was originally broken
in our system and can't be built ATM. Other distros such as ArchLinux
don't support GTK anymore (see https://git.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/avidemux#n64)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/video/avidemux/default.nix [hunk 1]
```diff
@@ -12,122 +13,73 @@
...
+    done
+    ln -s "$out/bin/avidemux3_${default}" "$out/bin/avidemux"
+
+    fixupPhase
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = http://fixounet.free.fr/avidemux/;
+    description = "Free video editor designed for simple video editing tasks";
+    maintainers = with maintainers; [ viric abbradar ma27 ];
```
Comment (Ma27) [inline pkgs/applications/video/avidemux/default.nix]: @viric @abbradar is it ok to keep you as maintainers as I changed way to much in this derivation :)

**Comment by abbradar** [comment]
My original motivation for this complexity was to split Avidemux over several derivations so that one can build GTK and Qt plugins separately from core libraries. However given that there is only one backend now which people would likely use (Qt 5) it doesn't seem to worth it now. Thanks!

P.S.: I'm okay with keeping me as a maintainer.

**Comment by Ma27** [comment]
great @abbradar .
Btw is there anything that keeps us from merging? :)

**Comment by abbradar** [comment]
@Ma27 Nope, I'll test and merge if everything is okay today. Thanks for your work!

**Pull request merged**